DokanNet/HelperMethods.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DokanNet
+{
+    public class HelperMethods
+    {
+        /// <summary>
+        /// Matches zero or more characters until encountering and matching the final . in the name.
+        /// </summary>
+        const Char DOS_STAR = '<';
+        /// <summary>
+        /// Matches any single character or, upon encountering a period or end
+        /// of name string, advances the expression to the end of the set of
+        /// contiguous DOS_QMs.
+        /// </summary>
+        const Char DOS_QM = '>';
+        /// <summary>
+        /// Matches either a period or zero characters beyond the name string.
+        /// </summary>
+        const Char DOS_DOT = '"';
+        /// <summary>
+        /// Matches zero or more characters.
+        /// </summary>
+        const Char ASTERISK = '*';
+        /// <summary>
+        /// Matches a single character.
+        /// </summary>
+        const Char QUESTION_MARK = '?';
+
+        private static Char[] CharsThatMatchEmptyStringsAtEnd = new Char[] { DOS_DOT, DOS_STAR, ASTERISK };
+        /// <summary>
+        /// check whether Name matches Expression
+        /// Expression can contain "?"(any one character) and "*" (any string)
+        /// when IgnoreCase is true, do case insenstive matching
+        /// </summary>
+        /// <seealso cref="http://msdn.microsoft.com/en-us/library/ff546850(v=VS.85).aspx"/>
+        /// <param name="Expression">Matching pattern</param>
+        /// <param name="Name">Name to test</param>
+        /// <param name="IgnoreCase">Ignore case</param>
+        /// <returns></returns>

```

Comment:
- Have a capital letter at the beginning of sentences and point at the end.
- Explain more how this feature should be used, and when. Preferably with a reference to IDokanOperations.FindFilesWithPattern. To further understand how this should be used for the uninitiated, add some example code.
- Do not forget to write something in the `<return>` tag.